doc/contribute/guidelines.rst

```diff
@@ -429,6 +431,8 @@ following exceptions:
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
 
+When there are differences between the guidelines above and the rules enforced
+by the ``clang-format`` configuration, the guidelines above take precedence.
```

Comment:
@keith-zephyr can we please move those lines to a separate PR so we can unblock this one? This is pretty unrelated to the actual max line length.

Reply:
@jfischer-no resolved his comment, but I've modified the language to be explicit that the documented coding style takes precedence over any code formatter.s.